test/sql/test_array_fn/R/test_arrays_zip

```diff
@@ -0,0 +1,564 @@
+-- name: test_arrays_zip @mac
+SELECT arrays_zip([1, 2], ['1b', '2b']) AS result;
+-- result:
+[{"col1":1,"col2":"1b"},{"col1":2,"col2":"2b"}]
+-- !result
+SELECT arrays_zip([1, 2], ['1b', null, '3b']) AS result;
+-- result:
+[{"col1":1,"col2":"1b"},{"col1":2,"col2":null},{"col1":null,"col2":"3b"}]
+-- !result
+SELECT arrays_zip([1, 2, 3], ['a', 'b', 'c'], [10, 20, 30]) AS result;
+-- result:
+[{"col1":1,"col2":"a","col3":10},{"col1":2,"col2":"b","col3":20},{"col1":3,"col2":"c","col3":30}]
```

Comment:
is it a struct? how about introducing a tuple type instead of struct for these scenarios?

tuple type is more efficient then struct.